lib/syscall_shim/uk_syscall_binary.c

```diff
@@ -90,6 +92,10 @@ void ukplat_syscall_handler(struct uk_syscall_ctx *usc)
 	ukplat_coutk(prsyscallbuf, (__sz) prsyscalllen);
 #endif /* CONFIG_LIBSYSCALL_SHIM_STRACE */
 
+#if CONFIG_LIBPOSIX_PROCESS_SIGNAL
+	uk_signal_deliver(usc);
```

Comment:
Ditto. But otherwise don't forget to also surround the forward declaration above in `#if`'s.